src/test/scala/strawman/collection/test/Test.scala

```diff
@@ -469,6 +469,8 @@ class StrawmanTest {
     val xs9t: mutable.HashMap[Int, String] = xs9
     val xs10 = xs1 ++ xs2
     val xs11: immutable.SortedMap[String, Int] = xs10
+    val xs12 = xs11.collect({ case (k, v) if k.length == v => (v, k) }: scala.PartialFunction[(String, Int), (Int, String)])
```

Comment:
Just to be clear, this cannot work in 2.12 because support for overloaded methods with `PartialFunction` formal arguments was only introduced in 2.13 via https://github.com/scala/scala/commit/9cd3becde5bb7c37af64ad5ac1bd15aa4fafd1b4. I'm investigating the error there (it should behave the same way as it does for `Function1`). It should work in Dotty where the same changes were made in https://github.com/lampepfl/dotty/pull/2015.